src/Telegram.php

```diff
@@ -965,4 +1001,24 @@ public function isRunCommands()
     {
         return $this->run_commands;
     }
+
+    /**
+     * Switch to enable running getUpdates without a database
+     *
+     * @param bool $enable
+     */
+    public function useGetUpdatesWithoutDatabase($enable = true)
+    {
+        $this->getupdates_without_database = $enable;
+    }
+
+    /**
+     * Return last update id
+     *
+     * @return int
+     */
+    public function getLastUpdateId()
```

Comment:
Good point, no idea why I put it there

Reply:
Oh wait! Probably to let user access last update id inside hook script, the purpose of that would be setting a signal handler (to prevent not marking updates as handled)

```php
        function signal_handler(Longman\TelegramBot\Telegram $telegram, $signo = null)
        {
            $last_update_id = $telegram->getLastUpdateId();
            if ($last_update_id > 0) {
                Longman\TelegramBot\Request::getUpdates(
                    [
                        'offset'  => $last_update_id + 1,
                        'limit'   => 1,
                    ]
                );
            }

            exit(PHP_EOL);
        }
```

Reply:
I didn't fully understand what you meant there. Where would this `signal_handler` method be called?
Also, the `$signo` parameter isn't being used, what is that?

Reply:
Okay, lets say the bot is heavily used and has to handle 40 updates at once, when bot crashes in the middle of processing they aren't marked as handled, next time script is run they are re-processed.

About `$signo`, for some reason that function refused to work without that parameter there... I tested it on linux subsystem since **pcntl** doesn't work on Windows.

Also that function is just an example, it can also be used to register shutdown function which should script errors and when called in `catch` block can also work after exceptions.

Reply:
Aaah, right! You're using the pcntl signal handling, thought it was just a custom function you call somewhere.

Ok. I'm just wondering if there is a cleaner way to do this. Also, we'll have to document this somehow with the extra code required to register the signal handler.

Reply:
Probably a wiki article mentioning and example code will suffice.